src/plugins/cosmos/components/modals/Staking/Staking.tsx

```diff
@@ -1,54 +1,119 @@
 import { useModal } from 'context/ModalProvider/ModalProvider'
-import { bnOrZero } from 'lib/bignumber/bignumber'
+import { BigNumber, bnOrZero } from 'lib/bignumber/bignumber'
 
-import { ClaimConfirmRouter } from './views/ClaimConfirmRouter'
+import { ClaimBroadcast } from './views/ClaimBroadcast'
+import { ClaimConfirm } from './views/ClaimConfirm'
 import { Overview } from './views/Overview'
 import { Stake } from './views/Stake'
-import { StakeConfirmRouter, StakingConfirmProps } from './views/StakeConfirmRouter'
+import { StakeBroadcast } from './views/StakeBroadcast'
+import { StakeConfirm } from './views/StakeConfirm'
 import { Unstake } from './views/Unstake'
-import { UnstakeConfirmRouter } from './views/UnstakeConfirmRouter'
+import { UnstakeConfirm } from './views/UnstakeConfirm'
 
 export enum StakingAction {
   Stake = 'stake',
   Unstake = 'unstake',
   Overview = 'overview',
   Claim = 'claim'
 }
+export type StakingModalLocation = {
+  cryptoAmount: BigNumber
+  assetId: CAIP19
+  fiatRate: BigNumber
+  apr: string
+}
 
-type StakingModalProps = {
+export type StakingModalProps = {
   assetId: CAIP19
-  action: StakingAction
 }
 
 export enum StakeRoutes {
+  Overview = '/overview',
   Stake = '/stake',
-  Unstake = '/unstake',
   StakeConfirm = '/stake/confirm',
+  StakeBroadcast = '/stake/broadcast',
+  Unstake = '/unstake',
   UnstakeConfirm = '/unstake/confirm',
-  Overview = '/stake/overview',
-  Claim = '/claim'
+  UnstakeBroadcast = '/unstake/broadcast',
+  ClaimConfirm = '/claim/confirm',
+  ClaimBroadcast = '/claim/broadcast'
 }
 
+export const stakeSteps = [
+  { step: 0, path: StakeRoutes.Stake, label: 'Amount' },
+  { step: 1, path: StakeRoutes.StakeConfirm, label: 'Confirm' },
+  { step: 2, path: StakeRoutes.StakeBroadcast, label: 'Broadcast' }
+]
+
+export const unstakeSteps = [
+  { step: 0, path: StakeRoutes.Unstake, label: 'Amount' },
+  { step: 1, path: StakeRoutes.UnstakeConfirm, label: 'Confirm' },
+  { step: 2, path: StakeRoutes.UnstakeBroadcast, label: 'Broadcast' }
+]
+
+export const claimSteps = [
+  { step: 0, path: StakeRoutes.ClaimConfirm, label: 'Confirm' },
+  { step: 1, path: StakeRoutes.ClaimBroadcast, label: 'Broadcast' }
+]
+
 export const entries = [
+  StakeRoutes.Overview,
   StakeRoutes.Stake,
-  StakeRoutes.Unstake,
   StakeRoutes.StakeConfirm,
+  StakeRoutes.StakeBroadcast,
+  StakeRoutes.Unstake,
   StakeRoutes.UnstakeConfirm,
-  StakeRoutes.Overview
+  StakeRoutes.UnstakeBroadcast,
+  StakeRoutes.ClaimConfirm,
+  StakeRoutes.ClaimBroadcast
 ]
 
-const StakingModalContent = ({ assetId, action }: StakingModalProps) => {
-  const location = useLocation<StakingConfirmProps>()
+const StakingModalContent = ({ assetId }: StakingModalProps) => {
+  const location = useLocation<StakingModalLocation>()
   const history = useHistory()
   const isConfirmStep = matchPath(location.pathname, {
-    path: [StakeRoutes.StakeConfirm, StakeRoutes.UnstakeConfirm],
+    path: [StakeRoutes.UnstakeConfirm],
```

Comment:
You don't need to wrap this in an array and can just use the string directly if matching a single route, primitives are always king